src/components/GistsList.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import GistHeader from './GistHeader'
+import { fetchGists, createGist, fetchStarGist } from '../helpers'
+
+class GistList extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      gists: [],
+    }
+  }
+
+
+  componentDidMount() {
+    fetchGists().then((response) => {
+      const gists = response.data._embedded.gists
+      this.setState({ gists })
+    })
+  }
+
+  handleCreate() {
+    createGist().then(response => console.log(response))
+    alert('gist created')
+  }
+
+  renderList() {
+    if (this.state.gists) {
+      return (
+        <div className="control">
+          <ul>
+            {this.state.gists.map(gist => (
+              <li key={gist.id} className="label">
+                <Link to={`/gist/${gist.id}`}>
+                  {gist.description}
+                </Link>
+              </li>))}
+          </ul>
+        </div>
+      )
+    }
+    return <span>Loading ... </span>
+  }
+
+  renderStarList() {
+    if (fetchStarGist()) {
```

Comment:
I think this should be replaced with different kind of check:
`if (this.state.gists.length > 0)`

Reply:
replaced